core/controlplane/config/templates/stack-template.json

```diff
@@ -328,20 +385,137 @@
       },
       "Type": "AWS::EC2::NetworkInterface"
     },
+    {{end}}
+    {{if $etcdInstance.EIPManaged}}
+    "{{$etcdInstance.EIPLogicalName}}": {
+      "Properties": {
+        "Domain": "vpc"
+      },
+      "Type": "AWS::EC2::EIP"
+    },
+    {{end}}
     {{if not $.EtcdDataVolumeEphemeral}}
-    "InstanceEtcd{{$etcdIndex}}ebs": {
+    "{{$etcdInstance.EBSLogicalName}}": {
       "Properties": {
-          "AvailabilityZone": "{{ $etcdInstance.SubnetAvailabilityZone}}",
+          "AvailabilityZone": "{{$etcdInstance.SubnetAvailabilityZone}}",
           "Size": "{{$.EtcdDataVolumeSize}}",
           {{if gt $.EtcdDataVolumeIOPS 0}}
           "Iops": "{{$.EtcdDataVolumeIOPS}}",
           {{end}}
-          "VolumeType": "{{$.EtcdDataVolumeType}}"
+          "VolumeType": "{{$.EtcdDataVolumeType}}",
+          "Tags": [
+            {
+              "Key": "kube-aws:owner:role",
+              "Value": "etcd"
+            },
+            {{if $etcdInstance.EIPManaged}}{
+              "Key": "{{$.Etcd.EIPAllocationIDTagKey}}",
+              "Value": {{$etcdInstance.EIPAllocationIDRef}}
+            },{{end}}
+            {{if $etcdInstance.NetworkInterfaceManaged}}{
+              "Key": "{{$.Etcd.NetworkInterfaceIDTagKey}}",
+              "Value": {{$etcdInstance.NetworkInterfaceIDRef}}
+            },{{end}}
+            {
+              "Key": "{{$.Etcd.AdvertisedHostNameTagKey}}",
+              "Value": {{$etcdInstance.AdvertisedHostNameRef}}
+            }
+          ]
       },
       "Type": "AWS::EC2::Volume"
     },
     {{end}}
-    "InstanceEtcd{{$etcdIndex}}": {
+    "{{$etcdInstance.LogicalName}}": {
+      "Type": "AWS::AutoScaling::AutoScalingGroup",
+      "Properties": {
+        "HealthCheckGracePeriod": 600,
+        "HealthCheckType": "EC2",
+        "LaunchConfigurationName": {
+          "Ref": "{{$etcdInstance.LaunchConfigurationLogicalName}}"
+        },
+        "MaxSize": "1",
+        "MetricsCollection": [
+          {
+            "Granularity": "1Minute"
+          }
+        ],
+        "MinSize": "1",
+        "Tags": [
+          {
+            "Key": "KubernetesCluster",
+            "PropagateAtLaunch": "true",
+            "Value": "{{$.ClusterName}}"
+          },
+          {
+            "Key": "Name",
+            "PropagateAtLaunch": "true",
+            "Value": "{{$.ClusterName}}-{{$.StackName}}-kube-aws-etcd"
```

Comment:
nit but would anyone like to append the index of an etcd node so that the instance name looks like `<cluster-name>-<stack-name>-kube-aws-etcd-<index>` which provides more visibility?